### ISSUE TITLE
The piwik documentation for setting up the database don't work for any mysql packages in nixpkgs.

### ISSUE THREAD

**Issue by dalaing** [NixOS/nixpkgs#27703]
## Issue description

The documentation mentions that we can use ALTER USER to set up the authentication for the piwik user.

That can't be done with MySQL 5.7 because of #24400, and that command is not supported in MySQL 5.5 or any of the versions of mariadb in nixpkgs.

### Steps to reproduce

Try to set up the database by following the documentation.

I had more luck following this:
https://github.com/NixOS/nixpkgs/issues/25266#issuecomment-302988959
but I'm having other issues after that.

I'm pretty keen to see this up and running, so let me know if there's other things I should be trying :)

## Technical details

* System: NixOS 17.03.1606.6bf7904140 (Gorilla)
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs versions: nixpkgs:"17.03.1606.6bf7904140" unstable:"17.09pre111304.5328102325"
* Sandboxing enabled: build-use-sandbox = false

**Comment by joachifm** [comment]
cc @florianjacob

**Comment by florianjacob** [comment]
The ALTER USER line just shouldn't be there. Damn, wrong paste, that is from my personal setup for securing the mysql root user. It is not needed for the piwik database setup at all. Could you test the setup with just skipping that command?

(unrelated to piwik itself: Note that the nixos mysql root user needs no authentication whatsoever by default and you either want to use socket authentication or set a password. While the update on mariadb 10.2 is not done / mysql 5.7 is not usable, you can have socket authentication with `UPDATE mysql.user SET plugin='unix_socket' WHERE user='root';` instead of `ALTER USER` on mariadb 10.1 (that's what I use) and mysql 5.5.)

**Comment by dalaing** [comment]
As far as I can tell, these are chunks of setup code that work for different DBs:

```
package = pkgs.mariadb;
initialScript = pkgs.writeText "piwik-sql-init" ''
  INSTALL PLUGIN unix_socket SONAME 'auth_socket';
  CREATE DATABASE piwik;
  CREATE USER 'piwik'@'localhost' IDENTIFIED WITH unix_socket;
  GRANT ALL PRIVILEGES ON piwik.* TO 'piwik'@'localhost';
  '';
```

```
package = pkgs.mysql55;
initialScript = pkgs.writeText "piwik-sql-init" ''
  INSTALL PLUGIN auth_socket SONAME 'auth_socket.so';
  CREATE DATABASE piwik;
  CREATE USER 'piwik'@'localhost' IDENTIFIED WITH auth_socket;
  GRANT ALL PRIVILEGES ON piwik.* TO 'piwik'@'localhost';
  '';
```

I can PR a change to the docs in a few days time, but I figured I'd put them here in case I forget or in case someone needs this between now and then.

**Comment by florianjacob** [comment]
Wow, the plugins are named auth_socket and unix_socket, making things incompatible, great. 😆

I'd appreciate a PR updating the docs accordingly, please link it here so I can cross-check it. 👍

You know about the problem that `initialScripts` are only executed on the very first database start, and never after? Or isn't that a problem with NixOPs? While I use `initialScripts` myself as a memory aid and for recreating the machine, I'd probably not mention it in the docs to avoid confusion for users who already have a database up and running and *then* want to install piwik, which is probably the majority.

**Comment by florianjacob** [comment]
(@dalaing note that the automatically generated “referenced“ backlink here won't trigger a notification for participants in this issue, I only saw #27820 because I still had this tab open. 😉 )

**Comment by florianjacob** [comment]
@dalaing can you close this manually? This was not closed automatically when #27820 was merged.
